rust/crd/src/listener.rs

```diff
@@ -149,30 +167,45 @@ pub fn get_kafka_listener_config(
             .insert(KafkaListenerName::Client, KafkaListenerProtocol::Plaintext);
     }
 
-    if kafka_security.tls_internal_secret_class().is_some() {
-        // 4) If internal tls is required we expose INTERNAL as SSL
+    // INTERNAL
+    if kafka_security.has_kerberos_enabled() {
```

Comment:
See 2d4feac. I've combined them, but have specified the secure port rather than the function, as this was place that internal tls was enforced for kerberos, rather than relying on `self.tls_internal_secret_class()` having already been set.